### PR TITLE
feat: add root gauges and refactor workflows

### LINKED ISSUE #180: feat: add root gauges
currently only has the child gauges

### PR DESCRIPTION
**Pull request by gosuto-inzasheru**
closes #180 

~~wip, currently only refactors the workflows~~

### CODE REVIEW

**Comment by gosuto-inzasheru** [comment]
- adds root gauges to the address book in the same way as there are gauges now (dumps a json, is a property of the addy book object, etc.)
- the generation of new (root) gauges and pools jsons is now its own separate workflow, to not overcrowd the weekly diffs

**Comment by Tritium-VLK** [comment]
Can we merge ecosystem balances first so I can review against that delta?

File: bal_addresses/pools_gauges.py
Comment (Tritium-VLK) [inline bal_addresses/pools_gauges.py:80]: We should refactor this at some point to use the whole query system we just implemented.  If you'd prefer to get this merged can make a ticket to do it later to resolve this.

File: gen_pools_and_gauges.py
Comment (Tritium-VLK) [inline gen_pools_and_gauges.py:146]: extras are normally user inputs and outputs go in outputs. Just noticed this now, but it's a bit confusing because in most cases extras/ are user defined things and outputs are generated things have a different file per chain. It seems like it would make sense to align this, and I think a number of things are already using the current output format. Thoughts?

Comment (Tritium-VLK) [reply gen_pools_and_gauges.py:146]: Also I thought we were going to break core pools into a separate action with it's own delta, but it seems to me like this doesc ore pools in the pools_gauges run and deletes the .py file that did core pools.   Don't we want a separate script and a separate action/branch/pr?

**Review by Tritium-VLK — CHANGES_REQUESTED** [review]

**Comment by gosuto-inzasheru** [comment]
all good loints, no rush with merging this so will work on getting those changes in here

File: gen_pools_and_gauges.py
Comment (gosuto-inzasheru) [reply gen_pools_and_gauges.py:146]: c295c18

Comment (gosuto-inzasheru) [reply gen_pools_and_gauges.py:146]: > Also I thought we were going to break core pools into a separate action with it's own delta, but it seems to me like this doesc ore pools in the pools_gauges run and deletes the .py file that did core pools. Don't we want a separate script and a separate action/branch/pr?

for this, i indeed batched core pools generation with the other pools and gauges generation. _that_ batch however, is now a separate action (`generate_pools_gauges.yaml`), independent of the general address book generation.

i think seeing the diffs on both all new pools and core pools could be useful actually, no?

Comment (Tritium-VLK) [reply gen_pools_and_gauges.py:146]: Pools there are tons of, it is very noisy and permissionless. 

Core pools are permissioned and very important.  For this reason I think it is important that we separate out the core pools deltas by themselves adn have a much more detailed review process there.

Comment (gosuto-inzasheru) [reply gen_pools_and_gauges.py:140]: 0aa4b22

File: .github/workflows/generate_addressbooks.yaml
Comment (Tritium-VLK) [inline .github/workflows/generate_addressbooks.yaml:27]: Transform deployments is the thing that generates addressbook.json, which is the thing that most people use.

It looks like you ru generate_addresses after each thing, but not transform_deployments.

Maybe in the end we need to just refactor gen_addresses and transform deployments.  For now it's a bit strange to be running one and not the other.  

Not sure if we should block this PR because of this, eveything will end up eventually consistent, but given the current setup it could take a couple weeks.

File: gen_pools_and_gauges.py
Comment (Tritium-VLK) [inline gen_pools_and_gauges.py:73]: Does it make sense to move this kind of logic to the subgraphs object?

**Review by Tritium-VLK — CHANGES_REQUESTED** [review]
Just a few more things to take alook at.  Mostly just think about the process query stuff.

I think we can rethink the action scheduling in another PR.

Once you do nore more onceover, the last thing I want to od is run all available actions on  this branch and make sure they all work right/render decent results. Once we merge we should run any new ones to also make sure they work.

File: gen_pools_and_gauges.py
Comment (gosuto-inzasheru) [reply gen_pools_and_gauges.py:73]: not really, this is very specific to the `gen_pools_and_gauges.py` workflow imo

File: .github/workflows/generate_addressbooks.yaml
Comment (gosuto-inzasheru) [reply .github/workflows/generate_addressbooks.yaml]: hmm ok good point. i shuffled the workflows in 389d983 a bit:
- `generate_addressbooks.yaml`: create jsons for pools, gauges, chains, and the mono addressbook
- `generate_permissions.yaml`: create permissions jsons
- `generate_core_pools.yaml`: create core pools json

Comment (Tritium-VLK) [reply .github/workflows/generate_addressbooks.yaml]: OK but addressbook.json + all the slashbooks (reverse nad forward) per chain all need to be up to date.  It seems like transform_deployments generates the addressbook and gen_addresses generates the flatbooks, and then all these other things generate artifacts that these 2 include.  This seems like it  will get messy.  Let me look at it a bit more.

**Review by Tritium-VLK — APPROVED** [review]

**Pull request merged**
